rpc/eth_api.go

```diff
@@ -313,13 +317,78 @@ func (e *PublicEthAPI) GetBlockByHash(hash common.Hash, fullTx bool) map[string]
 }
 
 // GetBlockByNumber returns the block identified by number.
-func (e *PublicEthAPI) GetBlockByNumber(blockNum rpc.BlockNumber, fullTx bool) map[string]interface{} {
-	return nil
+func (e *PublicEthAPI) GetBlockByNumber(blockNum rpc.BlockNumber, fullTx bool) (map[string]interface{}, error) {
+	value := blockNum.Int64()
+	block, err := e.cliCtx.Client.Block(&value)
+	if err != nil {
+		return nil, err
+	}
+	header := block.BlockMeta.Header
+
+	genesis, err := e.cliCtx.Client.Genesis()
+	if err != nil {
+		return nil, err
+	}
+	gasLimit := genesis.Genesis.ConsensusParams.Block.MaxGas
+
+	var gasUsed *big.Int
+	var transactions []interface{}
+
+	if fullTx {
+		// Populate full transaction data
+		transactions, gasUsed = convertTransactionsToRPC(e.cliCtx, block.Block.Txs,
+			common.BytesToHash(header.ConsensusHash.Bytes()), uint64(header.Height))
+	} else {
+		// TODO: Gas used not saved and cannot be calculated by hashes
+		// Return slice of transaction hashes
+		transactions = make([]interface{}, len(block.Block.Txs))
+		for i, tx := range block.Block.Txs {
+			transactions[i] = common.BytesToHash(tx.Hash())
+		}
+	}
+
+	return map[string]interface{}{
+		"number":           header.Height,
+		"hash":             header.ConsensusHash,
+		"parentHash":       header.LastBlockID.Hash,
+		"nonce":            nil, // PoW specific
+		"sha3Uncles":       nil, // No uncles in Tendermint
+		"logsBloom":        "",  // TODO: Complete with #55
+		"transactionsRoot": header.DataHash,
+		"stateRoot":        header.AppHash,
+		"miner":            header.ValidatorsHash,
+		"difficulty":       nil,
```

Comment:
Do we have anything we can provide for this?

Reply:
The reason it's as this is because transactions are ambiguous, but I can just create a type with transactions as ambiguous type.